packages/react-components/react-tree/library/src/components/TreeItem/useTreeItem.tsx

```diff
@@ -145,10 +148,57 @@ export function useTreeItem_unstable(props: TreeItemProps, ref: React.Ref<HTMLDi
     });
   });
 
+  const { findFirstFocusable } = useFocusFinders();
+
+  const handleTreeGridActionsKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
+    if (!event.currentTarget.contains(event.target as Node) || !treeItemRef.current) {
+      return;
+    }
+    switch (event.key) {
+      case treeDataTypes.ArrowLeft: {
+        // TODO: this should be included in requestTreeResponse navigation signature
+        treeItemRef.current?.focus();
+        return;
+      }
+      case treeDataTypes.ArrowUp: {
+        requestTreeResponse({
+          requestType: 'navigate',
+          event,
+          value,
+          itemType,
+          parentValue: undefined,
+          type: event.key,
+          target: treeItemRef.current,
+        });
+        return;
+      }
+      case treeDataTypes.ArrowDown: {
+        if (!isHTMLElement(event.target) || event.target.hasAttribute('aria-haspopup')) {
+          return;
+        }
+        requestTreeResponse({
+          requestType: 'navigate',
+          event,
+          value,
+          itemType,
+          parentValue: undefined,
+          type: event.key,
+          target: treeItemRef.current,
+        });
```

Comment:
The only difference I see between these requests and the ones in the main switch case is that the `parentValue` is undefined, is that significant?

Reply:
It is not relevant for this interaction, only if it was `ArrowLeft`, but for the sake of keeping everything similar, I'll add the proper value, good catch 😉